include/nlohmann/json.hpp

```diff
@@ -3486,31 +3489,32 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     access by reference
     @sa see @ref value() for access by value with a default value
 
-    @since version 1.0.0
+    @since version 1.0.0; template added in version 3.10.0
 
     @liveexample{The example below shows how object elements can be read and
     written using `at()`. It also demonstrates the different exceptions that
     can be thrown.,at__object_t_key_type}
     */
-    reference at(const typename object_t::key_type& key)
+    template < class KeyT, typename std::enable_if <
+                   !std::is_same<typename std::decay<KeyT>::type, json_pointer>::value&& (
+#if defined(JSON_HAS_CPP_17)
+                       std::is_same<typename std::decay<KeyT>::type, std::string_view>::value ||
+#endif
+                       std::is_convertible<KeyT, typename object_t::key_type>::value), int >::type    = 0 >
```

Comment:
Minor comment:

You can use ... instead of =0 and having to specify the type for enable-if statements in the template parameter list.

Personally I consider that to be more ergonomic to type and easier to read, because there are less non-relevant details presented.

Reply:
Can you provide an example? I always struggle with the syntax of `enable_if`...

Reply:
```cpp
    template < class KeyT, std::enable_if_t <
                         ! std::is_same_v<std::decay_t<KeyT>, json_pointer>
                   && (
#if defined(JSON_HAS_CPP_17)
                           std::is_same_v<std::decay_t<KeyT>, std::string_view>::value
#endif
                           || std::is_convertible_v<KeyT, typename object_t::key_type>
                       )> ... >
```

Reply:
`std::is_same_v` needs to be backported. I see if that makes sense in this PR.

Reply:
(`std::is_same_v` needs templated variables and does not work in C++11. But I added a helper `detail::is_key_type` that could remove a lot of clutter.)